import_ldraw.py

```diff
@@ -836,21 +836,41 @@ def create_model(self, context, scale):
 
             # Select all the mesh
             for cur_obj in objects:
+                bpy.ops.object.select_all(action='DESELECT')
                 cur_obj.select = True
                 bpy.context.scene.objects.active = cur_obj
-                if bpy.ops.object.mode_set.poll():
-
-                    # Change to edit mode
-                    bpy.ops.object.mode_set(mode='EDIT')
-                    bpy.ops.mesh.select_all(action='SELECT')
 
-                    # Add small gaps between each brick
-                    bpy.ops.transform.resize(value=(0.99, 0.99, 0.99))
-                    if bpy.ops.object.mode_set.poll():
+                gapWidth = 0.007
+                objScale = cur_obj.scale * scale
+                dim = cur_obj.dimensions
+                
+                # Checks whether the object isn't flat in a certain direction
+                # to avoid division by zero.
+                # Otherwise, the scale factor is proportional to the inverse of
+                # the dimension so that the mesh shrinks a fixed distance
+                # (determined by the gap_width and the scale of the object)
+                # in every direction, creating a uniform gap.
+                if dim.x != 0:
+                    facX = 1 - 2 * gapWidth * abs(objScale.x) / dim.x
+                else:
+                    facX = 1
+                if dim.y != 0:
+                    facY = 1 - 2 * gapWidth * abs(objScale.y) / dim.y
+                else:
+                    facY = 1
+                if dim.z != 0:
+                    facZ = 1 - 2 * gapWidth * abs(objScale.z) / dim.z
+                else:
+                    facZ = 1
 
-                        # Go back to object mode
-                        bpy.ops.object.mode_set(mode='OBJECT')
+                bpy.context.object.scale[0] *= facX
+                bpy.context.object.scale[1] *= facY
+                bpy.context.object.scale[2] *= facZ
 
+                bpy.ops.object.transform_apply(location=False, rotation=False,\
```

Comment:
The backslash is unnecessary. Also, because the first two parameters are named and [default to False](https://www.blender.org/api/blender_python_api_2_76_release/bpy.ops.object.html?highlight=transform_apply#bpy.ops.object.transform_apply), you should be able to simply reduce this line to

```python
bpy.ops.object.transform_apply(scale=True)
```